MODULE.bazel

```diff
@@ -57,7 +57,7 @@ use_repo(bazel_lib_toolchains, "yq_windows_amd64")
 bazel_dep(name = "gazelle", version = "0.35.0", dev_dependency = True, repo_name = "bazel_gazelle")
 bazel_dep(name = "buildifier_prebuilt", version = "6.4.0", dev_dependency = True)
 bazel_dep(name = "bazel_skylib_gazelle_plugin", version = "1.5.0", dev_dependency = True)
-bazel_dep(name = "rules_go", version = "0.41.0", dev_dependency = True)
+bazel_dep(name = "rules_go", version = "0.44.0", dev_dependency = True)
```

Comment:
Are the updates needed?

Reply:
They resolve warnings about the version specified being lower than the resolved one, so I think they are a no-op in terms of compat. But I can revert this if you'd prefer

Reply:
I guess it might be nice to have the non-bzlmod versions match, although version resolution with WORKSPACE is a "fun time"

Reply:
If it's not needed lets not do it in this PR. I'm trying to do some in https://github.com/aspect-build/rules_js/pull/1537 atm

Reply:
ok, reverted